cmd/tempo-vulture/main.go

```diff
@@ -127,8 +134,14 @@ func main() {
 			startTime, seed = selectPastTimestamp(startTime, now, interval, tempoRetentionDuration)
 
 			// Don't attempt to read on the first itteration if we can't reasonably
-			// expect the write loop to have fired yet.
-			if seed.Before(actualStartTime.Add(tempoWriteBackoffDuration)) {
+			// expect the write loop to have fired yet.  Double the duration here to
+			// avoid a race.
+			if seed.Before(actualStartTime.Add(2 * tempoWriteBackoffDuration)) {
+				continue
+			}
+
+			// Do the same for the startTime we've just selected
+			if seed.Before(startTime.Add(tempoWriteBackoffDuration)) {
```

Comment:
does this matter? if startTime > actualStartTime it means that vulture has been running longer than the retention period which means there shouldn't be a race condition?

Reply:
I've adjusted the search and read loops to use the simpler logic.  Thanks for the suggestion.